code/game/gamemodes/changeling/powers/shriek.dm

```diff
@@ -46,4 +46,5 @@
 		L.on = 1
 		L.break_light_tube()
 	empulse(get_turf(user), 2, 4, 1)
+	user.SetConfused(0) //Prevents confusion via EMP'ing your own items. Allows for better use of EMP while attempted to escape combat.
 	return 1
```

Comment:
very iffy on this implementation.
This will cleanse confuses that weren't caused by EMPs. I recommend adding a `TRAIT_CONFUSION_IMMUNE`, adding that before the EMP, and removing it afterwards. 
If you have any questions feel free to DM/ping me.

Reply:
Just going to scrap this to be honest. The easiest way to not deal with this is just avoid carrying flashes and it doesnt make much sense when looking at the ability by itself to prevent external causes from having an effect.